### PR TITLE
update cmake flags and add one for Cray FORTRAN

### LINKED ISSUE #1242: update cmake flags adding Cray FORTRAN
## Description
keep the flags in sync between ioda-converters and ioda

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

update compiler flags keeping in sync with IODA

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #1242

### CODE REVIEW

File: cmake/compiler_flags_Intel_Fortran.cmake [hunk 1]
```diff
@@ -1,40 +1,44 @@
...
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation nor
+# does it submit to any jurisdiction.
 
 ####################################################################
 # FLAGS COMMON TO ALL BUILD TYPES
 ####################################################################
 
-set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -traceback")
+set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -D_REAL8_")
```
Comment (BenjaminRuston) [inline cmake/compiler_flags_Intel_Fortran.cmake:13]: @srherbener do you know why the *normal* Fortran flags would have `-g -traceback` for iodaconv these should be in `DEBUG` no?

File: cmake/compiler_flags_Intel_Fortran.cmake [hunk 2]
```diff
@@ -1,40 +1,44 @@
-# (C) Copyright 2020-2020 NOAA/NWS/NCEP/EMC
+# (C) Copyright 2009-2016 ECMWF.
```
Comment (BenjaminRuston) [inline cmake/compiler_flags_Intel_Fortran.cmake:1]: @srherbener and @PatNichols what should we put here?

File: cmake/compiler_flags_GNU_Fortran.cmake [hunk 3]
```diff
@@ -1,41 +1,40 @@
...
+# does it submit to any jurisdiction.
 
 ####################################################################
 # FLAGS COMMON TO ALL BUILD TYPES
 ####################################################################
 
-set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -fbacktrace")
-
-if(CMAKE_Fortran_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
-    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch")
```
Comment (BenjaminRuston) [inline cmake/compiler_flags_GNU_Fortran.cmake:13]: @srherbener  and @PatNichols  do we want to have `-fallow-agrument-mismatch` ?

File: cmake/compiler_flags_Cray_Fortran.cmake [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# (C) Copyright 2023 JCSDA
```
Comment (BenjaminRuston) [inline cmake/compiler_flags_Cray_Fortran.cmake:1]: @PatNichols please make consistent

**Review by PatNichols — APPROVED** [review]
Looks good.

File: cmake/compiler_flags_Intel_Fortran.cmake (hunk 1 above)
Comment (srherbener) [reply cmake/compiler_flags_Intel_Fortran.cmake]: I agree with applying the `-traceback` option for only the debug build. The tracebacks from the optimized builds are usually unintelligible.

File: cmake/compiler_flags_Intel_Fortran.cmake (hunk 2 above)
Comment (srherbener) [reply cmake/compiler_flags_Intel_Fortran.cmake]: I think the copyright notices look good now. Thanks for fixing those!

**Review by fcvdb — APPROVED** [review]
Thanks Ben

File: cmake/compiler_flags_GNU_Fortran.cmake (hunk 3 above)
Comment (srherbener) [reply cmake/compiler_flags_GNU_Fortran.cmake:13]: I agree with removing the `-fallow-argument-mismatch` option. I can't think of a case where you would want to allow argument mismatching (of which I think could lead to some quite difficult debugging efforts).

**Review by srherbener — APPROVED** [review]
Thanks for fixing this! I tested ioda-bundle on my Mac, and ioda-bundle built successfully and all of the ioda and ioda-converters tests passed.

**Pull request merged**